### ISSUE TITLE
Block#getStateForNeighborUpdate -> something else

### ISSUE THREAD

**Issue by quat1024** [asiekierka/MinecraftPOMF#1935]
This isn't a simple getter function, and many blocks perform side effects in this method.

For example, `BeehiveBlock` calls `angerBees` if there is fire next to the beehive, but otherwise just delegates to `super`. `CactusBlock` schedules a scheduled tick that makes the cactus break if it is adjacent to something. Many mounted blocks like signs break if they're not against something, many waterloggable blocks schedule fluid ticks, etc.

Basically this method is named like a getter, but it's not really a getter, it's more like a function that happens to return something. I can't think of a good name though.

**Comment by liach** [comment]
`updateStateForNeighbor`? (Didn't read code yet)
